irohad/model/execution/command_executor.hpp

```diff
@@ -79,16 +97,39 @@ namespace iroha {
                            ametsuchi::WsvQuery &queries,
                            const std::string &creator_account_id) = 0;
 
-      logger::Logger log_;
+      /**
+       * String name of the command
+       * @return name of the command for which executor is invoked
+       */
+      virtual std::string commandName() const noexcept = 0;
+
+      /**
+       * Creates Error out of given error message
+       * @param error_message what message should be displayed to the user
+       * @return Error object which can be used for result construction
+       */
+      expected::Error<ExecutionError> makeExecutionError(
+          const std::string &error_message) const noexcept;
+
+      /**
+       * Return execution result with an error with given message if predicate
+       * is false
+       * @param predicate - condition which determines result state
+       * @param error_message - string which will be part of the error
+       * @return ExecutionResult with empty value if predicate is true, Error
+       * with error message otherwise
+       */
+      ExecutionResult errorIfNot(bool predicate,
```

Comment:
The name `predicate` is confusing. A predicate is a function that returns bool, but not a bool.